package/network/config/wifi-scripts/files/lib/wifi/mac80211.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/sh
+
+append DRIVERS "mac80211"
+
+mac80211_get_board_phy_name() (
+	local path="$1"
+	local fallback_phy=""
+
+	__check_phy() {
+		local val="$1"
+		local key="$2"
+		local ref_path="$3"
+
+		json_select "$key"
+		json_get_vars path
+		json_select ..
+
+		[ "${ref_path%+*}" = "$path" ] && fallback_phy=$key
+		[ "$ref_path" = "$path" ] || return
+
+		echo "$key"
+		exit
+	}
+
+	json_load_file /etc/board.json
+	json_for_each_item __check_phy wlan "$path"
+	[ -n "$fallback_phy" ] && echo "${fallback_phy}.${path##*+}"
+)
+
+mac80211_rename_board_phy_by_path() {
+	local path="$1"
+
+	local new_phy="$(mac80211_get_board_phy_name "$path")"
+	[ -z "$new_phy" -o "$new_phy" = "$phy" ] && return
+
+	iw "$phy" set name "$new_phy"
+}
+
+mac80211_rename_board_phy_by_name() (
+	local phy="$1"
+	local suffix="${phy##*.}"
+	[ "$suffix" = "$phy" ] && suffix=
+
+	json_load_file /etc/board.json
+	json_select wlan
+	json_select "${phy%.*}" || return
+	json_get_vars path
+
+	prev_phy="$(iwinfo nl80211 phyname "path=$path${suffix:++$suffix}")"
+	[ -n "$prev_phy" ] || return
+
+	[ "$prev_phy" = "$phy" ] && return
+
+	iw "$prev_phy" set name "$phy"
+)
+
+mac80211_rename_phy() {
+	local phy path macaddr
+
+	config_get phy "$1" phy
+	[ -n "$phy" ] && {
+		[ -d /sys/class/ieee80211/$phy ] && return
```

Comment:
Isn't this doing something similar compared to line 52?
`[ "$prev_phy" = "$phy" ] && return`

Line 62: does phy of said name already exist, then skip. else: issue renaming by name.
Line 52: does prev_phy already have the expected name then skip. else issue renaming by name using iw.

I assume Line 62 will never encounter a phy name already existing unless it's the correct phy name. Else we will run into race-conditions due to name conflicts. example: phy1 exists but is supposed to be named wl1, phy0 exists but is supposed to be named phy1.

Should we consider dropping 52 or 62? (or better save then sorry?)

Reply:
I have added line 62 as an early exit for repeated runs of the renaming code. It is true that the check in line 52 probably can't return true if the one in line 62 isn't. I'll wait for @nbd168's judgement on this.

Reply:
@nbd168 Mind to share a verdict? :)